### ISSUE TITLE
Kiota Serialization objects completely unusable

### ISSUE THREAD

**Issue by stephajn** [microsoftgraph/msgraph-sdk-dotnet#2442]
### Describe the bug

I am working with retrieving CustomSecurityAttributes for a Graph User.
In version 5.44.0 of the library, the values of these attributes are returned as JsonElements which I can retrieve property data from.

When I updated to 5.48.0, the JsonElements were replaced with UntypeObject values instead. 

1. This is a breaking change!
2. The UntypedObject class is completely unusable because the dictionary it returns is an IDictionary<string, UntypedNode>. When I retrieve any value from it, All I get back is an UntypedNode object. I can't cast this to UntypedBoolean, UntypedString etc. In addition, just trying to call GetValue() on the value just throws a NotImplementedException rather than actually giving me a value.

I had to revert back to 5.44.0.

### Expected behavior

I expect to not have to deal with such breaking changes like this in a minor release version.

I expect to be able to properly retrieve values from the UntypedObject in a much more intuitive way than has been provided.

### How to reproduce

Just try to retrieve custom security attributes for a Graph User that has them set.

### SDK Version

5.48.0

### Latest version known to work for scenario above?

5.44.0

### Known Workarounds

Don't upgrade to 5.48.0 and stay on 5.44.0 instead.

**Comment by quinterojose** [comment]
This is very unfortunate. I just spent all afternoon troubleshooting this issue. The following code was working fine before the update:

```
string[]? existingOrganizationNodeIds = [];
if ((existingUser?.AdditionalData.TryGetValue($"extension_{extensionAppId}_organizationNodeIds", out var value) ?? false)
    && value is JsonElement jsonElement)
{
    existingOrganizationNodeIds = jsonElement.EnumerateArray().Select(item => item.ToString()).ToArray();
}
```

Serializing the UntypedArray object to a string using KiotaJsonSerializer then deserializing back to a string[] using JsonSerializer solves the issue but man!

Here's the new code:

```
string[]? existingOrganizationNodeIds = [];
if ((existingUser?.AdditionalData.TryGetValue($"extension_{extensionAppId}_organizationNodeIds", out var value) ?? false)
    && value is UntypedArray untypedArray)
{
    var tempJson = KiotaJsonSerializer.SerializeAsString(untypedArray);
    existingOrganizationNodeIds = JsonSerializer.Deserialize<string[]>(tempJson);
}
```

Here's another similar issue reported https://github.com/microsoft/kiota-serialization-json-dotnet/issues/212

**Comment by petrhollayms** [comment]
Related: #2459

**Comment by mejohnm** [comment]
Not sure if I should create another thread but the same thing is happening when getting the checklist collection from the Planner Task Details. I am using Microsoft.Graph version 5.55 and also tried the latest beta since I also need the Recurrence which only beta gets.


![image](https://github.com/microsoftgraph/msgraph-sdk-dotnet/assets/41702555/bae42c44-cda4-4047-98c0-087a93b5acef)

**Comment by Cezus** [comment]
Any updates?

**Comment by Kraeuterbuddah** [comment]
Any updates?

**Comment by mariussm** [comment]
Yeah, this was not fun to work with. Here is my workaround: https://goodworkaround.com/2024/10/07/working-around-custom-security-attribute-limitations-in-net-graph-sdk/

**Comment by ashley-w-brazier-cloudm** [comment]
Any update on this at all?

**Comment by MartinM85** [comment]
Do you have any examples how are your custom security attributes defined? 

At least me, I'm able to read the value of CSA.

![image](https://github.com/user-attachments/assets/72b0168a-c180-4aa0-9fb3-1bb9beb32a8a)

SDK code:

![image](https://github.com/user-attachments/assets/7a5f2e4e-d534-4c44-95d2-f9feb4ff62cd)

**Comment by cgjedrem** [comment]
I am using this serialization which is very easy and handy for the newest Microsoft Graph  5.74.0. Since the newest changes the .AdditionalDataProperty is as many mentioned, here an UntypedObject drom Kiota package. To serialize it to your own object do the following. Here I have Organizations from Microsoft graph

```
// Define the task for each group's additional data
var task = groups
    .Select(async group =>
    {
        if (!group.AdditionalData.TryGetValue("someProperty", out var additionalData))
            return null;

        // Here we tell the compiler that additionalData is an UntypedObject 
        // so that serialization works correctly
        if (additionalData is not UntypedObject untypedObject) 
            return null;

        // Serialize to JSON string
        var serializedData = await KiotaJsonSerializer.SerializeAsStringAsync(untypedObject);

        // Deserialize to your strongly typed object
        var schema = JsonConvert.DeserializeObject<YourObject>(serializedData);
        return schema;
    });

// Process all additional data for your groups
var companyInfoResponse = await Task.WhenAll(task);

// Perform further LINQ transformations
var companyInfo = companyInfoResponse
    .Where(company => company != null)
    .GroupBy(company => company!.ID)
    .Select(company => company.First())
    .FirstOrDefault();
```
Basically you have to tell the compiler that your additional data is an UntypedObject before Serializing it with KiotaJsonSerializer
